EIPS/eip-6353.md

```diff
@@ -0,0 +1,333 @@
+---
+eip: 6353
+title: Charity token
+description: Extension of EIP-20 Token that can be partially donated to a charity project
+author: Aubay <blockchain-team@aubay.com>, BOCA Jeabby (@bjeabby1507), EL MERSHATI Laith (@lth-elm), KEMP Elia (@eliakemp)
+discussions-to: https://ethereum-magicians.org/t/erc20-charity-token/12617
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-05-13
+requires: 20
+---
+
+## Abstract
+
+An extension to the [EIP-20](./eip-20.md) standard token that allows Token owners to donate to some charity organization. The following standard allows for the implementation of the standard token to enforce charity donation by default. This standard is an extension of the [EIP-20](./eip-20.md) token to provide transfers to a third-party charity during transactions. This standard also allows token that support [EIP-20](./eip-20.md) interface, to have a standardized way of signalling charity information, registered by the token contract.
+
+## Motivation
+
+The initial idea is to allow Token owners to easily donate passively, in order to facilitate contributions to non-profit organizations and facilitate integration by giving a generalized implementation. Users can make an impact with their token and can contribute to achieving sustainable blockchain development. It is the possibility to donate simply with micro-donation during transactions. Projects can easily retrieve charity donations addresses and rate for a given [EIP-20](./eip-20.md) token, token holders can compare minimum rate donation offers allowed by token contract owners.
+
+## Specification
+
+**EIP-20 compliant contract MAY implement this ERC for automatizing Mirco payment during transactions.**
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+Owner of the contract **MAY**, after review, register charity address in `whitelistedRate` and set globally a default rate of donation. To register the address, the rate **MUST** not be null.
+
+Token holders **MAY** choose and specify a default charity address from `_defaultAddress`, this address **SHOULD** be different from the null address for the donation to be activated.
+
+Token holders **MAY** choose a donation rate different from the default one, individually for a specific charity address, by modifying the rate in `_donation`.
+
+Donations are calculated as a percentage of the amount of token transfered, this percentage is added to the initial amount of token transferred. `_feeDenominator()` is overridable but defaults to 10000, meaning the rate is specified in basis points by default but **MAY** be customizable just as the `_defaultRate()` that default to 10.
+
```

Comment:
You haven't actually specified how the donations are withdrawn from the account. You'll want to mention:

 * `transfer` and `transferFrom` must reduce the sender's balance by whatever formula and increase the donation recipient's balance by the same amount.
 * That donations are in addition to the transferred amount.
 * That these functions emit two `Transfer` events (pretty sure it'd break EIP-20 if it didn't.)
 * What happens if the account has a sufficient balance to cover the transfer but not the donation. Does the whole transfer revert? Does the transfer go through without a donation? Is it up to the token implementer to decide?